### PR TITLE
chore: Continue populating `replications_specs.*.id` value even when calling new API

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-262014

Populating existing `replication_spec.*.id` for replica set and symmetric sharded clusters is relevant to avoid breaking the use case of cloud backup schedule with currently uses this value ([example](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_backup_schedule#example-usage---create-a-cluster-with-cloud-backup-enabled-with-snapshot-distribution)).
As part of this PR existing tests related to cloud backup schedule are now unskipped and working.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by EspenAlbert — APPROVED** [review]
Nice!

File: internal/service/advancedcluster/resource_advanced_cluster.go [hunk 1]
```diff
@@ -579,6 +585,26 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
...
 }
 
+// getReplicationSpecIDsFromOldAPI returns the id values of replication specs coming from old API. This is used to populate old replication_specs.*.id attribute avoiding breaking changes.
+// In the old API each replications spec has a 1:1 relation with each zone, so ids are returned in a map from zoneName to id.
+func getReplicationSpecIDsFromOldAPI(ctx context.Context, projectID, clusterName string, connV220231115 *admin20231115.APIClient) (map[string]string, error) {
+	clusterOldAPI, _, err := connV220231115.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+	if apiError, ok := admin20231115.AsError(err); ok {
+		readErrorMsg := "error reading  advanced cluster with 2023-02-01 API (%s): %s"
+		if apiError.GetErrorCode() == "ASYMMETRIC_SHARD_UNSUPPORTED" {
+			tflog.Info(ctx, fmt.Sprintf(readErrorMsg, clusterName, err))
```
Comment (lantoli) [inline internal/service/advancedcluster/resource_advanced_cluster.go]: are these logs shown in Prod?   not sure about logging non-error or non-warning info

**Review by lantoli — APPROVED** [review]

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 2]
```diff
@@ -1405,13 +1405,23 @@ func configShardedNewSchema(orgID, projectName, name, instanceSizeSpec1, instanc
 	`, orgID, projectName, name, instanceSizeSpec1, instanceSizeSpec2, diskIopsSpec1, diskIopsSpec2)
 }
 
-func checkShardedNewSchema(instanceSizeSpec1, instanceSizeSpec2, diskIopsSpec1, diskIopsSpec2 string) resource.TestCheckFunc {
+func checkShardedNewSchema(instanceSizeSpec1, instanceSizeSpec2, diskIopsSpec1, diskIopsSpec2 string, replicationSpecOldIDDefined bool) resource.TestCheckFunc {
```
Comment (maastha) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go]: if the use-case of asymmetric cluster is the only one for which the `replicationSpecOldIDDefined` flag will be false, should we rename it to `isAsymmetricCluster` or something?

**Review by maastha — APPROVED** [review]

File: internal/service/advancedcluster/resource_advanced_cluster.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster.go]: yes they would be shown, path that is being logged is not too relevant so simply removing

File: internal/service/advancedcluster/resource_advanced_cluster_test.go (hunk 2 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster_test.go]: like the idea, renamed

**Comment by AgustinBettati** [comment]
failing tests are due to out of capacity and not related to these changes: `TestAccClusterAdvancedClusterConfig_replicationSpecsAnalyticsAutoScaling`, `TestAccClusterAdvancedCluster_replicaSetAWSProvider`

**Review by maastha — APPROVED** [review]

**Pull request merged**